docs/guides/2_using_aligned_on_your_app.md

```diff
@@ -0,0 +1,111 @@
+# Using Aligned on your app
+
+You can find an example of the full flow of using Aligned on your app 
+in the [ZKQuiz example](../../examples/zkquiz). 
+
+This example shows a sample app that generates a SP1 proof 
+that a user knows the answers to a quiz, then submits the proof 
+to Aligned for verification. 
+Finally, it includes a smart contract that verifies that a proof 
+was verified in Aligned and mints an NFT.
+
+## Steps
+
+### Step 1 - Write your ZK Proof
+
+Write your ZK proof using any of the proving systems supported by Aligned.
+For this example, we use the SP1 proving system.
+
+You can find the example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) 
+as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) 
+that generates it in the [ZKQuiz example](../../examples/zkquiz) folder.
+
+### Step 2 - Write your smart contract
+
+Write your smart contract that verifies the proof was verified in Aligned.
+For this, you will need a way to check that the proven program is your own.
+
+The aligned cli provides a way for you to get the verification key commitment 
+without actually generating and submitting a proof.
+
+You can do this by running the following command:
+
+```bash
+aligned get-vk-commitment --input <path_to_input_file>
+```
+
+For SP1 you would use the elf of the program as the input file.
+
+You can find the example of the smart contract that verifies the proof was verified in Aligned
+in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+Note that the contract has and checks that the verification key commitment is the same as the program elf.
+```solidity
+require(elfCommitment == provingSystemAuxDataCommitment, "ELF does not match");```
```

Comment:
remove extra ` ``` `
```suggestion
require(elfCommitment == provingSystemAuxDataCommitment, "ELF does not match");
```